### DISCUSSION TITLE
How to disable services of an package with adb/su shell ?

### DISCUSSION THREAD

**Discussion by xDoge26** [lihenggui/blocker#164] [Q&A]
Hi, i want to disable some google services with pm disable or intent firewall ? and how do i check running service of an package ?
This is the list i want to disable 

```
com.google.android.gms/com.google.android.gms.common.stats.GmsCoreStatsService
com.google.android.gms/com.google.android.gms.gcm.GcmService
com.google.android.gms/com.google.android.gms.chimera.PersistentDirectBootAwareApiService
com.google.android.gms/com.google.android.gms.chimera.GmsIntentOperationService
com.google.android.gms/com.google.android.gms.chimera.PersistentIntentOperationService
```

**Comment by lihenggui** [comment]
The command is `pm disable [PackageName/ComponmentName]`

First, you need to open adb shell. `adb shell`
Then gain the su permission. `su`
Then type the command `pm disable com.google.android.gms/com.google.android.gms.common.stats.GmsCoreStatsService`

If you want to enable components, just replace `disable` to `enable` in the command. 
This is called PM mode, there is no API for the intent firewall to add rules.
